examples/ex02_cksum/test/ChecksumRTL_test.py

```diff
@@ -106,20 +109,20 @@ def setup_class( cls ):
   # [setup_method] will be called by pytest before executing each class method.
   # See pytest documetnation for more details.
   def setup_method( s, method ):
-    s.vcd_file_name = ""
-
-    import sys
-    if hasattr( sys, '_pymtl_dump_vcd' ):
-      if sys._pymtl_dump_vcd:
-        s.vcd_file_name = "{}.{}".format( s.DutType.__name__, method.__name__ )
+    s.current_test_method_name = method.__name__
 
   # [teardown_method] will be called by pytest after executing each class method.
   # See pytest documetnation for more details.
   def teardown_method( s, method ):
-    s.vcd_file_name = ""
+    s.current_test_method_name = ""
 
   def run_sim( s, th, max_cycles=1000 ):
 
+    s.vcd_file_name = ""
+
+    if s.__class__.cmdline_opts["dump_vcd"]:
```

Comment:
why is it s.__class__?

Reply:
it seems radically different from the old one and I'm not sure how the new one works.

Reply:
See this line: https://github.com/cornell-brg/pymtl3/blob/846187d57d32a4fa9537ff6afe8dd136d8751a1c/pytest_plugin/pytest_pymtl3.py#L43
That line dynamically injects `cmdline_opts` as an attribute of the test class. We can also inject that to test functions but since most test functions will have `cmdline_opts` in their arguments that might not be necessary.

Reply:
they are the same?
https://github.com/cornell-brg/pymtl3/blob/301fec576e327114468ec1fb81b31c57303b6d6b/conftest.py#L28

Reply:
`cmdline_opts['dump_vcd']` has the same value as the previous `dump_vcd` fixture?

Reply:
the fixture dump_vcd actually comes up with the string, but here the run_sim function assembles the string

Reply:
I think Yanghui did this because he was not able to get the fixture value. I'll change this since now we can use the fixture.